### PR TITLE
Move benchmarks to separate repository

### PR DESCRIPTION
**Pull request by cmnrd**
This PR removes all the benchmarks from this repository, as was discussed in https://github.com/lf-lang/lingua-franca/issues/347. The benchmarks are now located in the new repo https://github.com/lf-lang/benchmarks. I adjusted the CI workflows such that the LF workflow call the benchmark test workflow in lf-lang/benchmarks.

Important: after merging this PR to master, [this line](https://github.com/lf-lang/benchmarks/blob/175ef0247ba945bf7facc12171696f217a5f6dfd/.github/workflows/benchmark-tests.yml#L12) needs to be deleted in the benchmark workflow.

Note that this PR is based on #993 and should only be merged after it.

### CODE REVIEW

File: example/C/src/MQTT/MQTTPhysical.lf [hunk 1]
```diff
@@ -61,7 +61,7 @@
  */
 target C {
     // Must use threaded implementation so schedule is thread safe.
-    threads: 1,
+    workers: 1,
```
Comment (lhstrh) [inline example/C/src/MQTT/MQTTPhysical.lf:64]: Remove?

File: example/C/src/MQTT/MQTTPublisher.lf [hunk 2]
```diff
@@ -6,7 +6,7 @@
  * @author Edward A. Lee
  */
 target C {
-    threads: 1  // This makes sure the threaded runtime is used, giving access to the mutex.
+    workers: 1  // This makes sure the threaded runtime is used, giving access to the mutex.
```
Comment (lhstrh) [inline example/C/src/MQTT/MQTTPublisher.lf:9]: Remove?

File: example/C/src/MQTT/MQTTSubscriber.lf [hunk 3]
```diff
@@ -7,7 +7,7 @@
  * @author Edward A. Lee
  */
 target C {
-    threads: 1  // This makes sure the threaded runtime is used, giving access to the mutex.
+    workers: 1  // This makes sure the threaded runtime is used, giving access to the mutex.
```
Comment (lhstrh) [inline example/C/src/MQTT/MQTTSubscriber.lf:10]: Remove?

File: example/C/src/ROS/ROSBuiltInSerialization.lf [hunk 4]
```diff
@@ -24,7 +24,7 @@
  target CCpp {
     cmake: true,  // Only CMake is supported
     cmake-include: "include/CMakeListsExtension.txt",
-    threads: 1,
+    workers: 1,
```
Comment (lhstrh) [inline example/C/src/ROS/ROSBuiltInSerialization.lf:27]: Remove?

File: example/C/src/ReflexGame.lf [hunk 5]
```diff
@@ -12,7 +12,7 @@
  * @author Marten Lohstroh
  */
 target C {
-    threads: 1,
+    workers: 1,
```
Comment (lhstrh) [inline example/C/src/ReflexGame.lf:15]: Remove?

File: example/C/src/ReflexGame/ReflexGameTest.lf [hunk 6]
```diff
@@ -1,5 +1,5 @@
 target C {
-    threads: 1,
+    workers: 1,
```
Comment (lhstrh) [inline example/C/src/ReflexGame/ReflexGameTest.lf:2]: Remove?

File: example/C/src/TrainDoor/TrainDoor.lf [hunk 7]
```diff
@@ -1,4 +1,4 @@
-target C {threads: 1, keepalive: true};
+target C {workers: 1, keepalive: true};
```
Comment (lhstrh) [inline example/C/src/TrainDoor/TrainDoor.lf:1]: Remove property?

File: example/Cpp/src/Patterns/FullyConnected_00_Broadcast.lf [hunk 8]
```diff
@@ -8,7 +8,7 @@
  */
 
 target Cpp {
-    threads: 1
+    workers: 1
```
Comment (lhstrh) [inline example/Cpp/src/Patterns/FullyConnected_00_Broadcast.lf:11]: Remove?

File: example/Cpp/src/Patterns/FullyConnected_01_Addressable.lf [hunk 9]
```diff
@@ -10,7 +10,7 @@
  */
 
 target Cpp {
-    threads: 1
+    workers: 1
```
Comment (lhstrh) [inline example/Cpp/src/Patterns/FullyConnected_01_Addressable.lf:13]: Remove?

File: example/Cpp/src/Patterns/MatrixConnectedRowsAndColumns.lf [hunk 10]
```diff
@@ -4,7 +4,7 @@
 // dimension. Nodes are organized in Rows which are grouped to form the matrix.
 
 target Cpp {
-    threads: 1
+    workers: 1
```
Comment (lhstrh) [inline example/Cpp/src/Patterns/MatrixConnectedRowsAndColumns.lf:7]: Remove?

File: org.lflang.tests/src/org/lflang/tests/Configurators.java [hunk 11]
```diff
@@ -46,24 +46,35 @@
...
+     * do not distinguish threaded and unthreaded runtime, the number
+     * of workers is set to 1.
      *
      * @param test The test to configure.
      * @return True if successful, false otherwise.
      */
     public static boolean useSingleThread(LFTest test) {
-        test.context.getArgs().setProperty("threads", "0");
+        test.context.getArgs().setProperty("threading", "false");
+        test.context.getArgs().setProperty("workers", "1");
```
Comment (lhstrh) [inline org.lflang.tests/src/org/lflang/tests/Configurators.java:61]: Isn't the workers property unnecessary/meaningless when `threading` is set to false?

File: org.lflang/src/org/lflang/generator/c/CGenerator.xtend [hunk 12]
```diff
@@ -378,8 +378,8 @@ class CGenerator extends GeneratorBase {
                 if (action.origin == ActionOrigin.PHYSICAL) {
                     // If the unthreaded runtime is requested, use the threaded runtime instead
                     // because it is the only one currently capable of handling asynchronous events.
-                    if (targetConfig.threads < 1) {
-                        targetConfig.threads = 1
+                    if (!targetConfig.threading) {
+                        targetConfig.threading = true
```
Comment (lhstrh) [inline org.lflang/src/org/lflang/generator/c/CGenerator.xtend:382]: This is meaning-preserving, but I'm not sure automatically enabling the multi-threaded library works in all scenarios (particularly bare-iron ones).

File: test/C/src/concurrent/AsyncCallbackDrop.lf [hunk 13]
```diff
@@ -3,7 +3,7 @@
 // This test will not work with the unthreaded C target because that target
 // does not implement any mutex protecting the event queue.
 target C {
-	threads: 1,
+	workers: 1,
```
Comment (lhstrh) [inline test/C/src/concurrent/AsyncCallbackDrop.lf:6]: Remove?

File: test/C/src/concurrent/AsyncCallbackReplace.lf [hunk 14]
```diff
@@ -3,7 +3,7 @@
 // This test will not work with the unthreaded C target because that target
 // does not implement any mutex protecting the event queue.
 target C {
-	threads: 1,
+	workers: 1,
```
Comment (lhstrh) [inline test/C/src/concurrent/AsyncCallbackReplace.lf:6]: Remove?

File: test/C/src/concurrent/CompositionThreaded.lf [hunk 15]
```diff
@@ -1,7 +1,7 @@
 // This test connects a simple counting source to tester
 // that checks against its own count.
 target C {
-	threads: 1,
+	workers: 1,
```
Comment (lhstrh) [inline test/C/src/concurrent/CompositionThreaded.lf:4]: Remove?

File: test/C/src/concurrent/DeadlineThreaded.lf [hunk 16]
```diff
@@ -3,7 +3,7 @@
 // are sent after a big enough delay to violate the deadline.
 target C {
     timeout: 3 sec,
-    threads: 1
+    workers: 1
```
Comment (lhstrh) [inline test/C/src/concurrent/DeadlineThreaded.lf:6]: Remove?

File: test/C/src/concurrent/DeadlineHandledAboveThreaded.lf [hunk 17]
```diff
@@ -1,6 +1,6 @@
 // Test a deadline where the deadline violation produces
 // an output and the container reacts to that output.
-target C {threads: 1};
+target C {workers: 1};
```
Comment (lhstrh) [inline test/C/src/concurrent/DeadlineHandledAboveThreaded.lf:3]: Remove?

File: test/C/src/concurrent/DeterminismThreaded.lf [hunk 18]
```diff
@@ -1,5 +1,5 @@
 target C {
-    threads: 1
+    workers: 1
```
Comment (lhstrh) [inline test/C/src/concurrent/DeterminismThreaded.lf:2]: Remove?

File: test/C/src/multiport/MultiportFromBank.lf [hunk 19]
```diff
@@ -2,7 +2,7 @@
  // Here, the bank is smaller than the width of the sending port.
 target C {
     timeout: 2 sec,
-    threads: 4,
+    workers: 4,
```
Comment (lhstrh) [inline test/C/src/multiport/MultiportFromBank.lf:5]: Generally speaking, it seems odd to select a specific number of threads for tests. Should we just slash all of these and leave it up to the runtime?

File: test/C/src/concurrent/DoubleReactionThreaded.lf [hunk 20]
```diff
@@ -4,7 +4,7 @@
 target C {
     timeout: 10 sec,
     fast: true,
-    threads: 1
+    workers: 1
```
Comment (lhstrh) [inline test/C/src/concurrent/DoubleReactionThreaded.lf:7]: I'll stop flagging these in the interest of reducing spam...

**Review by lhstrh — APPROVED** [review]
These changes look good to me! I think that in all tests where we had `threads: 1`, we should just remove the property (since `threading` is `on` by default). More generally, the specific numbers of workers `>1` given in tests seems rather arbitrary, so I would also suggest to remove those, provided that the C runtime, too, can sensibly select a default number of workers.

**Comment by cmnrd** [comment]
This PR is actually about moving the benchmarks. It only happens to be based on #993 because this way it already includes the necessary changes in the benchmark files.

We should continue the discussion regarding the workers property in #993.

**Pull request merged**